In [126]:
import pandas as pd

In [127]:
outlets = pd.read_csv("Data/QSR_outlet_dataset_cleaned.csv", keep_default_na= False)
socioeconomic = pd.read_csv("Data/QSR_socioeconomic_dataset_cleaned.csv")

In [128]:
print(f"QSR outlet row and column: {outlets.shape}")
print(f"Socio-Economic row and column: {socioeconomic.shape}")

QSR outlet row and column: (2816, 17)
Socio-Economic row and column: (20, 9)


In [129]:
outlets.head()

,Outlet ID,Brand,Region,Province,Province_Shorthand,City,Menu_Category,Price_Range,Customer_Rating,Review_Count,Latitude,Longitude,Location_Type,Address,Source,Price_Range_Cleaned,Price_Range_Imputed
0,1,McDonald's,Valle d'Aosta,Aosta,AO,Aosta,Burgers,EUR 1-10,3.8,3974,45.73953,7.345432,Commercial Area,"Aosta, 11100 Aosta, Aosta Valley",https://www.mcdonalds.it/elenco-ristoranti,EUR 1-10,False
1,2,McDonald's,Liguria,Genova,GE,Genova,Burgers,EUR 1-10,3.7,2244,44.40684,8.936673,City Center,Via Venti Settembre 1,https://www.mcdonalds.it/elenco-ristoranti,EUR 1-10,False
2,3,McDonald's,Liguria,Genova,GE,Genova,Burgers,EUR 1-10,3.9,2170,44.41372,8.879941,Shopping Mall,"Via Paolo Mantovani, 17, 16149 Genova GE",https://www.mcdonalds.it/elenco-ristoranti,EUR 1-10,False
3,4,McDonald's,Liguria,Genova,GE,Genova,Burgers,EUR 1-10,2.9,303,44.41724,8.921427,Transport Hub,"Piazza Acquaverde, 16134 Genova GE",https://www.mcdonalds.it/elenco-ristoranti,EUR 1-10,False
4,5,McDonald's,Liguria,Genova,GE,Genova,Burgers,EUR 1-10,3.8,3290,44.41141,8.928781,Transport Hub,"Via di Sottoripa, 127R/133, 16124 Genova GE",https://www.mcdonalds.it/elenco-ristoranti,EUR 1-10,False


In [130]:
socioeconomic.head()

,Region,Province_Count,Region_Area,Population,Population Density,Disposable Income per capita,Total Tourism Flow,Foreign Tourism Flow,Median Income
0,Lombardia,12,23863.65,10065694,421.8,27243,41795073.0,26300388.0,32485
1,Valle d'Aosta,1,3260.90,122554,37.6,25152,3692878.0,1439669.0,32900
2,Piemonte,8,25387.07,4255006,167.6,24416,14410448.0,7180603.0,30267
3,Trentino-Alto Adige,2,13605.50,1090818,80.2,28130,55227131.0,33544804.0,37563
4,Veneto,7,18407.42,4857460,263.9,24103,71896863.0,49799037.0,32272


In [106]:
#Checking Data Types

outlets.info()

<class 'pandas.DataFrame'>
RangeIndex: 2816 entries, 0 to 2815
Data columns (total 17 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Outlet ID            2816 non-null   int64  
 1   Brand                2816 non-null   str    
 2   Region               2816 non-null   str    
 3   Province             2816 non-null   str    
 4   Province_Shorthand   2816 non-null   str    
 5   City                 2816 non-null   str    
 6   Menu_Category        2816 non-null   str    
 7   Price_Range          2816 non-null   str    
 8   Customer_Rating      2816 non-null   float64
 9   Review_Count         2816 non-null   int64  
 10  Latitude             2816 non-null   float64
 11  Longitude            2816 non-null   float64
 12  Location_Type        2816 non-null   str    
 13  Address              2816 non-null   str    
 14  Source               2816 non-null   str    
 15  Price_Range_Cleaned  2816 non-null   str    
 16 

In [107]:
#Checking Data Types

socioeconomic.info()

<class 'pandas.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 9 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Region                        20 non-null     str    
 1   Province_Count                20 non-null     int64  
 2   Region_Area                   20 non-null     float64
 3   Population                    20 non-null     int64  
 4   Population Density            20 non-null     float64
 5   Disposable Income per capita  20 non-null     int64  
 6   Total Tourism Flow            20 non-null     float64
 7   Foreign Tourism Flow          20 non-null     float64
 8   Median Income                 20 non-null     int64  
dtypes: float64(4), int64(4), str(1)
memory usage: 1.5 KB


In [108]:
# Checking missing vlaues for QSR outlet

(outlets == "").sum()

Outlet ID               0
Brand                   0
Region                  0
Province                0
Province_Shorthand      0
City                    0
Menu_Category           0
Price_Range            63
Customer_Rating         0
Review_Count            0
Latitude                0
Longitude               0
Location_Type           0
Address                 0
Source                  0
Price_Range_Cleaned     0
Price_Range_Imputed     0
dtype: int64

In [109]:
#Checking for missing values

socioeconomic.isna().sum()

Region                          0
Province_Count                  0
Region_Area                     0
Population                      0
Population Density              0
Disposable Income per capita    0
Total Tourism Flow              0
Foreign Tourism Flow            0
Median Income                   0
dtype: int64

In [110]:
# Summary statistics for the numeric variables of the QSR outlet dataset
outlets.describe()

,Outlet ID,Customer_Rating,Review_Count,Latitude,Longitude
count,2816.000000,2816.000000,2816.000000,2816.000000,2816.000000
mean,1408.500000,3.871591,1856.995028,43.795357,11.155209
std,813.053504,0.407151,2964.480241,2.164930,2.230322
min,1.000000,1.000000,1.000000,36.901520,6.892710
25%,704.750000,3.700000,248.000000,41.915805,9.204194
50%,1408.500000,3.900000,1041.500000,44.796410,11.021635
75%,2112.250000,4.100000,2638.250000,45.491083,12.544072
max,2816.000000,5.000000,59200.000000,46.679270,18.197400


From this data we can know:
    <br />1, The average customer rating is 3.871591 out of 5 for all QSRs in Italy.
    <br>2, The average outlet has about 1858 reviews. But the review counts are very spread out. So, the median would be a better measure of a typical outlet
    <br>3, The coordinates are inside Italy. As Italy has Latitude between (35.5, 47.1) and Longitude between (6.6, 18.6)

In [111]:
# Summary statistics for numeric socioeconomic variables

socioeconomic.describe()

,Province_Count,Region_Area,Population,Population Density,Disposable Income per capita,Total Tourism Flow,Foreign Tourism Flow,Median Income
count,20.000000,20.000000,2.000000e+01,20.000000,20.000000,2.000000e+01,2.000000e+01,20.000000
mean,5.350000,15103.642000,2.947141e+06,177.885000,21767.100000,2.235850e+07,1.170911e+07,29163.300000
std,3.013566,7448.375038,2.509338e+06,112.313177,3843.639132,2.019318e+07,1.369814e+07,4658.277297
min,1.000000,3260.900000,1.225540e+05,37.600000,16173.000000,4.947860e+05,4.749700e+04,22225.000000
25%,3.500000,9167.117500,1.167826e+06,95.425000,17993.000000,7.776650e+06,1.668786e+06,24696.250000
50%,5.000000,14446.425000,1.691030e+06,158.300000,22732.000000,1.524733e+07,6.999274e+06,29490.500000
75%,7.250000,22586.345000,4.551555e+06,215.525000,24583.250000,3.983087e+07,1.463992e+07,32992.000000
max,12.000000,25832.390000,1.006569e+07,421.800000,28130.000000,7.189686e+07,4.979904e+07,37563.000000


From this result we can understand:
<br>1, How large Italian regions are in population. Italian regions are very different in population size. The average region has 2.95 million people but the median has only 1.69 million population size. Meaning, very few large regions are pulling the average up. Therefore, Most Italian regions are relatively small or medium-sized, while a few large regions contain a very high share of the national population.

2, How different are regions is population density. they vary very much in how crowded they are. some regions have low density, around 37.6 people per km2 while others have 442 people per km2. Therefore, QSR outlets may be more concentrated in dense regions because dense areas have more foot traffic, more urban centers, and more commercial activity.

3, There is also a clear income difference among the regions. Higher-income regions may support more QSR outlets, especially brands with higher price ranges, but this needs to be tested later with correlation and regression.

4, The maximum tourism flow is huge compared to the minimum. Some regions receive less than 500,000 tourist flows, while the highest region receives about 71.9 million. Telling us that, tourism is concentrated in small number of regions

5, Median income also varies strongly across regions. Some regions have much stronger household income conditions than others, which may influence restaurant demand, brand location strategy, and price-range distribution.

In [112]:
# Count outlets by QSR Brand

outlets["Brand"].value_counts()


Brand
McDonald's              790
La Piadineria           528
Autogrill               278
Old Wild West           243
Alice Pizza             180
Burger King             177
Roadhouse Restaurant    168
KFC                     153
Chef Express             99
Billy Tacos              69
Starbucks                48
Spontini                 25
Popeyes                  15
Spizzico                 14
Five Guys                10
Burgez                   10
Subway                    9
Name: count, dtype: int64

In [113]:
# The percentage share of outlets by Brand
outlets["Brand"].value_counts(normalize= True) * 100

Brand
McDonald's              28.053977
La Piadineria           18.750000
Autogrill                9.872159
Old Wild West            8.629261
Alice Pizza              6.392045
Burger King              6.285511
Roadhouse Restaurant     5.965909
KFC                      5.433239
Chef Express             3.515625
Billy Tacos              2.450284
Starbucks                1.704545
Spontini                 0.887784
Popeyes                  0.532670
Spizzico                 0.497159
Five Guys                0.355114
Burgez                   0.355114
Subway                   0.319602
Name: proportion, dtype: float64

From the above two results, we can see that the dataset is not evenly distributed across brand. Some brands have many outlets, while others have only a small number of observations. Meaning, overall results may be influenced more by the largest brands, and comparisons for smaller brands may be less reliable. Graph is present in the summary_visualization file

In [114]:
outlets["Menu_Category"].value_counts()

Menu_Category
Burgers          1230
Piadine           528
Bar/Café          377
Pizza             219
Fried Chicken     168
Steakhouse        168
Mexican            69
Coffee             48
Sandwiches          9
Name: count, dtype: int64

In [115]:
outlets["Menu_Category"].value_counts(normalize= True) * 100

Menu_Category
Burgers          43.678977
Piadine          18.750000
Bar/Café         13.387784
Pizza             7.776989
Fried Chicken     5.965909
Steakhouse        5.965909
Mexican           2.450284
Coffee            1.704545
Sandwiches        0.319602
Name: proportion, dtype: float64

In [116]:
# Count outlets by region

outlets["Region"].value_counts()


Region
Lombardia                799
Lazio                    414
Piemonte                 252
Veneto                   244
Emilia-Romagna           244
Toscana                  177
Campania                 125
Sicilia                  104
Liguria                   65
Marche                    63
Puglia                    62
Abruzzo                   56
Sardegna                  54
Friuli-Venezia Giulia     50
Calabria                  34
Trentino-Alto Adige       31
Umbria                    30
Valle d'Aosta              5
Basilicata                 4
Molise                     3
Name: count, dtype: int64

In [117]:
# Percentage share of outlets by region

outlets["Region"].value_counts(normalize=True) * 100


Region
Lombardia                28.373580
Lazio                    14.701705
Piemonte                  8.948864
Veneto                    8.664773
Emilia-Romagna            8.664773
Toscana                   6.285511
Campania                  4.438920
Sicilia                   3.693182
Liguria                   2.308239
Marche                    2.237216
Puglia                    2.201705
Abruzzo                   1.988636
Sardegna                  1.917614
Friuli-Venezia Giulia     1.775568
Calabria                  1.207386
Trentino-Alto Adige       1.100852
Umbria                    1.065341
Valle d'Aosta             0.177557
Basilicata                0.142045
Molise                    0.106534
Name: proportion, dtype: float64

From the above two results, we can see that outlets are concentrated in certain regions. Lombardia has the largest share of QSR outlets, while smaller regions such as Molise and Valle d'Aosta have very few outlets. This matters because regional concentration may be related to population size, tourism, income, and density.


In [118]:
# Count outlets by location type

outlets["Location_Type"].value_counts()


Location_Type
Commercial Area     999
Shopping Mall       910
Service Area        365
City Center         250
Transport Hub       225
Residential Area     67
Name: count, dtype: int64

In [ ]:
# Percentage share of outlets by location type

outlets["Location_Type"].value_counts(normalize=True) * 100

Location_Type
Commercial Area     35.475852
Shopping Mall       32.315341
Service Area        12.961648
City Center          8.877841
Transport Hub        7.990057
Residential Area     2.379261
Name: proportion, dtype: float64

These show which types of locations are most common for QSR outlets. Commercial areas and shopping malls dominate, suggesting that QSR outlets are mainly located in high-traffic consumer spaces.


In [120]:
# Count outlets by cleaned price range

outlets["Price_Range_Cleaned"].value_counts()


Price_Range_Cleaned
EUR 10-20    1328
EUR 1-10     1275
EUR 20-30     213
Name: count, dtype: int64

In [121]:
# Percentage share of outlets by cleaned price range

outlets["Price_Range_Cleaned"].value_counts(normalize=True) * 100


Price_Range_Cleaned
EUR 10-20    47.159091
EUR 1-10     45.276989
EUR 20-30     7.563920
Name: proportion, dtype: float64

The results with the price range show that most outlets are in the lower and middle price ranges. That tells us most of the outlets are affordable QSRs.

In [123]:
# Brand-level summary with outlet count, average rating, and average review count

outlets.groupby("Brand").agg(
    Outlet_Count=("Outlet ID", "count"),
    Average_Rating=("Customer_Rating", "mean"),
    Average_Review_Count=("Review_Count", "mean")
).sort_values(by="Average_Rating", ascending=False)


,Outlet_Count,Average_Rating,Average_Review_Count
Brand,,,
Popeyes,15,4.220000,417.866667
La Piadineria,528,4.089773,293.956439
Alice Pizza,180,4.064444,407.966667
Old Wild West,243,4.063786,1928.279835
Spizzico,14,4.057143,4287.571429
KFC,153,4.049673,1171.241830
Burger King,177,3.860452,2331.966102
Burgez,10,3.850000,441.400000
Roadhouse Restaurant,168,3.849405,1823.029762


This table shows the average customer rating and average review count combined with the number of outlets for each QSR brand. It helps compare how brands perform in terms of customer satisfaction and customer attention.

However, these averages should be interpreted carefully because the number of outlets is not the same for every brand. Brands with many outlets, such as McDonald's or La Piadineria, have more stable averages because they are based on many observations. Brands with only a few outlets can have averages that are strongly influenced by one or two locations.

In [124]:
# Region-level summary with outlet count, average rating, and average review count

outlets.groupby("Region").agg(
    Outlet_Count=("Outlet ID", "count"),
    Average_Rating=("Customer_Rating", "mean"),
    Average_Review_Count=("Review_Count", "mean")
).sort_values(by="Outlet_Count", ascending=False)


,Outlet_Count,Average_Rating,Average_Review_Count
Region,,,
Lombardia,799,3.829412,1625.804756
Lazio,414,4.004831,1794.391304
Piemonte,252,3.826190,1678.940476
Emilia-Romagna,244,3.807377,3114.061475
Veneto,244,3.811475,1772.545082
Toscana,177,3.848023,1976.847458
Campania,125,3.810400,1817.952000
Sicilia,104,3.895192,1473.086538
Liguria,65,3.847692,2532.553846


This table summarizes QSR outlets by region. It shows where outlets are concentrated and compares average customer rating and average review count across regions.

Regions with many outlets, such as Lombardia and Lazio, have more stable averages because they are based on many observations. Regions with very few outlets, such as Molise or Valle d'Aosta, should be interpreted more carefully because a small number of outlets can strongly affect the average rating or review count.